samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -52,8 +52,11 @@
   private final StreamManager streamManager;
   private final SystemAdmins systemAdmins;
 
+  protected final StreamGraphImpl streamGraph;
+
   public AbstractApplicationRunner(Config config) {
     super(config);
+    this.streamGraph = new StreamGraphImpl(this, config);
```

Comment:
I'd love to see some docs on the scope of the AppRunner - currently, appRunner.run(app) allows for multiple applications to be run with the same instance of the runner (atleast the api does not disallow it). 

With this change of instantiating the `StreamGraph` in the constructor, are we restricting the scope to running a single application? If not, how do we distinguish `StreamGraph`s of 2 different apps on the same runner

Reply:
@vjagadish agreed. Documentation will be added. However, I never thought that a single instance of AppRunner can be used for multiple different applications. The current implementation of LocalApplicationRunner actually has a bad combination of run with highlevel + runTask() with low level, which should not be the supported case at all. In addition, the internal state of LocalApplicationRunner is currently assumed to have a single application (i.e. processors are not separated by application instance names, counter down latch is not per application instance either). I would love to move to the direction that we enforce one application instance per AppRunner (i.e. essentially, we just create one application instance and runner is internally configured and attached for this application). I would prefer to move the doc effort of ApplicationRunner related change to a separate PR. This is the minimum required to get the current patch working.